data/mods/Xedra_Evolved/items/inventor/gun.json

```
@@ -0,0 +1,737 @@
+      "fields_max_intensity": 2,
+      "explosion": { "power": 225 }
+    },
+    "flags": [ "TRADER_AVOID", "BOMB" ]
+  },
+  {
+    "id": "sonic_gun",
+    "type": "GUN",
+    "name": { "str": "sonic gun" },
+    "description": "Small, square tube with pistol grip, that can emit a sonic waves, creating cavitations and tissue damage inside the living creatures.  Pretty weak as a weapon, using extremely high frequency sounds make it silent for most creatures you can met.  Would be fun to listen some music using it, but alas.",
+    "weight": "3490 g",
+    "volume": "4957 ml",
+    "longest_side": "998 mm",
+    "to_hit": -2,
+    "bashing": 3,
+    "material": [ "steel" ],
+    "symbol": "(",
+    "color": "magenta",
+    "skill": "rifle",
+    "ammo": "battery",
+    "range": 16,
+    "ranged_damage": { "damage_type": "pure", "amount": 14 },
+    "ammo_to_fire": 35,
+    "dispersion": 15,
+    "durability": 8,
+    "loudness": 0,
+    "flags": [ "NEVER_JAMS", "NO_SALVAGE", "NO_REPAIR", "NON_FOULING", "NEEDS_NO_LUBE", "TRADER_AVOID", "ELECTRONIC" ],
+    "pocket_data": [
+      {
+        "pocket_type": "MAGAZINE_WELL",
+        "rigid": true,
+        "flag_restriction": [ "BATTERY_LIGHT", "BATTERY_MEDIUM" ],
+        "max_contains_volume": "20 L",
+        "max_contains_weight": "20 kg",
+        "default_magazine": "medium_battery_cell"
+      }
+    ]
+  },
+  {
+    "id": "heater_gun",
+    "type": "TOOL",
+    "name": { "str": "\"Boiler\"", "str_pl": "\"Boilers\"" },
+    "description": "Rifle size tool, that can concentrate heat in a pretty huge range.  You can use it both to heat your food, or to heat the tissues of your enemies.",
+    "weight": "1908 g",
+    "volume": "3130 ml",
+    "longest_side": "1160 mm",
+    "to_hit": -2,
+    "bashing": 6,
+    "material": [ "steel" ],
+    "symbol": "|",
+    "color": "blue",
+    "charges_per_use": 50,
+    "ammo": [ "battery" ],
+    "flags": [ "TRADER_AVOID" ],
+    "//": "didn't add HOTPLATE and HEAT_FOOD to prevent an unpleasant 'i try to shoot a gun, but i need to accept i really want to cast a spell, and not heat my sandwich every time'",
+    "use_action": { "type": "cast_spell", "spell_id": "heat_expl_gun", "no_fail": true, "level": 0, "need_wielding": true },
+    "pocket_data": [
+      {
+        "pocket_type": "MAGAZINE_WELL",
+        "rigid": true,
+        "flag_restriction": [ "BATTERY_MEDIUM", "BATTERY_HEAVY" ],
+        "default_magazine": "medium_battery_cell"
+      }
+    ]
+  },
+  {
+    "id": "inventor_pistol",
+    "copy-from": "pistol_revolver",
+    "type": "GUN",
+    "name": { "str": "blocky pistol" },
+    "description": "A square abomination, that resemble a shape of a pistol, but with a collapsible outer magazine under the barrel.  Overengineered, this piece of metal is able to vary the size of chamber, which makes it able to shoot any round up to shotgun shells.  You spent really long time to make it fit into the common holster",
```

```suggestion
    "description": "A square abomination resembling the shape of a pistol, but with a collapsible outer magazine under the barrel.  This overengineered piece of metal is able to vary the size of chamber, which makes it able to shoot any round up to shotgun shells.  You spent a really long time to make it fit into a common holster.",
```
~~Varying the size of chamber wouldn't make it able to shoot rimfire ammo reee~~